Extend/Model/PaymentGateway.php

```diff
@@ -71,44 +68,37 @@ public function __construct()
      *
      * If you want to customize the descriptor, override this function.
      *
-     * @param string $sOrderId the order ID to get the descriptor from
-     *
-     * @return string
+     * @param Transaction $oTransaction the transaction to fill
+     * @param string      $sOrderId     the order ID to get the descriptor from
      *
      * @SuppressWarnings(PHPMD.Coverage)
      */
-    public function getDescriptor(string $sOrderId): string
+    private static function _addDescriptor(&$oTransaction, $sOrderId)
     {
         $shopId = Registry::getConfig()->getShopId();
         $shop = oxNew(Shop::class);
         $shop->load($shopId);
-        return substr($shop->oxshops__oxname->value, 0, 9) . " " . $sOrderId;
+        $oTransaction->setDescriptor(substr($shop->oxshops__oxname->value, 0, 9) . " " . $sOrderId);
     }
 
     /**
      * Returns a redirect object
      *
-     * @param Session $oSession
-     * @param string  $sShopUrl
+     * @param Session $oSession the current session
+     * @param string  $sShopUrl the shop's url
+     * @param string  $sSid     the query formated session id eg. "&sid=123456789"
      *
      * @return string
      *
      * @SuppressWarnings(PHPMD.Coverage)
      */
-    public function getRedirectUrls(Session $oSession, string $sShopUrl): Redirect
+    public static function getRedirectUrls($oSession, $sShopUrl, $sSid): Redirect
```

Comment:
Why not use the same logic with addRedirectUrls, always keep in mind that the logic and coding standard is consistent and easy to understand.

Reply:
👍